src/loaders/anndata-loaders/MatrixZarrLoader.js

```diff
@@ -0,0 +1,180 @@
+   * @param {string} shape Shape of the non-sparse output.
+   * @returns {Object} A { data: Float32Array } contianing the CellXGene matrix.
+   */
+  async _loadCSCSparseCellXGene(matrix, shape) {
+    const { store } = this;
+    const { data: cols } = await openArray({ store, path: `${matrix}/indptr`, mode: 'r' }).then(z => new Promise((resolve) => {
+      z.getRaw(null)
+        .then(resolve);
+    }));
+    const { data: rows } = await openArray({ store, path: `${matrix}/indices`, mode: 'r' }).then(z => new Promise((resolve) => {
+      z.getRaw(null)
+        .then(resolve);
+    }));
+    const { data: cellXGene } = await openArray({ store, path: `${matrix}/data`, mode: 'r' }).then(z => new Promise((resolve) => {
+      z.getRaw(null)
+        .then(resolve);
+    }));
+    const cellXGeneMatrix = new Float32Array(shape[0] * shape[1]).fill(0);
+    let col = 0;
+    cols.forEach((_, index) => {
+      const colStart = cols[index];
+      const colEnd = cols[index + 1];
+      for (let i = colStart; i < colEnd; i += 1) {
+        const val = cellXGene[i];
+        const row = rows[i];
+        cellXGeneMatrix[row * shape[1] + col] = val;
+      }
+      col += 1;
+    });
+    return { data: cellXGeneMatrix };
+  }
+
+  /**
+   * Class method for loading the cell x gene matrix.
+   * @returns {Promise} A promise for the zarr array contianing the cell x gene data.
+   */
+  async loadCellXGene() {
+    const { store, url } = this;
+    if (this.arr) {
+      return this.arr;
+    }
+    const {
+      options: { matrix },
+    } = this;
+    const res = await fetch(`${url}/${matrix.replace('.', '/')}/.zattrs`);
+    if (res.status !== 404) {
+      const zattrs = await res.json();
+      if (zattrs['encoding-type'] === 'csr_matrix') {
+        this.arr = this._loadCSRSparseCellXGene(matrix.replace('.', '/'), zattrs.shape);
+        return this.arr;
+      }
+      if (zattrs['encoding-type'] === 'csc_matrix') {
+        this.arr = this._loadCSCSparseCellXGene(matrix.replace('.', '/'), zattrs.shape);
+        return this.arr;
+      }
+    }
+    this.arr = openArray({ store, path: matrix.replace('.', '/'), mode: 'r' }).then(z => new Promise((resolve) => {
+      z.getRaw(null)
+        .then(resolve);
+    }));
+    return this.arr;
+  }
+
+  /**
+   * Class method for loading the genes list from AnnData.var.
+   * @returns {Promise} A promise for the zarr array contianing the gene names.
+   */
+  async loadGeneNames() {
+    if (this.geneNames) {
+      return this.geneNames;
+    }
+    const { genesFilter: genesFilterZarr } = this.options;
+    let genesFilter;
+    if (genesFilterZarr) {
+      const genesFilterArr = await openArray({
+        store: `${this.url}/${genesFilterZarr.replace('.', '/')}`,
+        mode: 'r',
+      });
+      const genesBufferCompressed = await genesFilterArr.store.getItem('0');
+      genesFilter = await genesFilterArr.compressor.decode(genesBufferCompressed);
```

Comment:
Ideally you shouldn't need to call `this.url` and instead use `this.store`. This is kind of a trick because calling `openArray` with a store being a string will make all paths relative. But it's important to note that `this.store` !== `genesFilterArr.store` in this code. To use the same store, you'd need something like:


```javascript
      const genesFilterArr = await openArray({
        store: this.store,
        path: genesFilterZarr.replace('.', '/'),
      });
      const genesBufferCompressed = await this.store.getItem(genesFilterZarr.replace('.', '/') + '/0');
      genesFilter = await genesFilterArr.compressor.decode(genesBufferCompressed);
```